### PR TITLE
Acceptance test to share a folder or file with group and a member performs actions on received share

### PR DESCRIPTION
**Pull request by jasson99**
## Description
Acceptance test to share a folder or file with group and a member performs actions on received share.The test to assert auto-completion is not present for a  group that is excluded from receiving shares is also added.

## Related Issue

- Fixes <issue_link>

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [x] Acceptance tests added
- [ ] Documentation ticket raised: <link> 

## Open tasks:

- [ ] ...

### CODE REVIEW

**Comment by jasson99** [comment]
@skshetry @dpakach @HariBhandari07

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>Trashbin</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/6352/

<p>













</p></details>

File: tests/acceptance/pageObjects/sharedWithMePage.js [hunk 1]
```diff
@@ -50,6 +50,14 @@ module.exports = {
         .waitForElementVisible(actionLocatorButton)
         .click(actionLocatorButton)
         .waitForOutstandingAjaxCalls()
+    },
+    assertSharedByDesiredUser: function (element, sharer) {
```
Comment (HariBhandari07) [inline tests/acceptance/pageObjects/sharedWithMePage.js]: documentation missing

Comment (jasson99) [reply tests/acceptance/pageObjects/sharedWithMePage.js]: Thankyou.I fixed that

**Comment by phil-davis** [comment]
@jasson99 now there are conflicts, probably from the other PR  of yours that I just merged.

File: tests/acceptance/pageObjects/sharedWithMePage.js [hunk 2]
```diff
@@ -68,6 +68,20 @@ module.exports = {
         .waitForElementVisible(actionLocatorButton)
         .click(actionLocatorButton)
         .waitForOutstandingAjaxCalls()
+    },
+    /**
+     * @param {string} element
+     * @param {string} sharer
+     *
+     *Asserts that the element(file/folder/resource) on the shared-with-me page is shared by the desired user
```
Comment (phil-davis) [inline tests/acceptance/pageObjects/sharedWithMePage.js]:
```suggestion
     * Asserts that the element(file/folder/resource) on the shared-with-me page is shared by the desired user
```

File: tests/acceptance/stepDefinitions/sharingContext.js [hunk 3]
```diff
@@ -228,6 +228,28 @@ Given('the administrator has excluded group {string} from sharing', async functi
...
+Given('the administrator has excluded group {string} from receiving shares', async function (group) {
+  const configList = await runOcc([
+    'config:list'
+  ])
+  const config = _.get(configList, 'ocs.data.stdOut')
+  const configParsed = JSON.parse(config)
+  const initialExcludedGroup = JSON.parse(_.get(configParsed, 'apps.files_sharing.blacklisted_receiver_groups') || '[]')
+  if (!initialExcludedGroup.includes(group)) {
+    initialExcludedGroup.push(group)
+    const resultGroupList = initialExcludedGroup.map((res) => '"' + res + '"')
```
Comment (skshetry) [inline tests/acceptance/stepDefinitions/sharingContext.js]:
```suggestion
    const resultGroupList = initialExcludedGroup.map((res) => `"${res}"`)
```

File: tests/acceptance/stepDefinitions/sharingContext.js [hunk 4]
```diff
@@ -228,6 +228,28 @@ Given('the administrator has excluded group {string} from sharing', async functi
   }
 })
 
+Given('the administrator has excluded group {string} from receiving shares', async function (group) {
+  const configList = await runOcc([
+    'config:list'
+  ])
+  const config = _.get(configList, 'ocs.data.stdOut')
+  const configParsed = JSON.parse(config)
+  const initialExcludedGroup = JSON.parse(_.get(configParsed, 'apps.files_sharing.blacklisted_receiver_groups') || '[]')
```
Comment (skshetry) [inline tests/acceptance/stepDefinitions/sharingContext.js]: Why does this require a parse? Is the string escaped?

Comment (skshetry) [reply tests/acceptance/stepDefinitions/sharingContext.js]: String of array is returned. Got it.

File: tests/acceptance/stepDefinitions/sharingContext.js [hunk 5]
```diff
@@ -228,6 +228,28 @@ Given('the administrator has excluded group {string} from sharing', async functi
...
+  if (!initialExcludedGroup.includes(group)) {
+    initialExcludedGroup.push(group)
+    const resultGroupList = initialExcludedGroup.map((res) => '"' + res + '"')
+    const resultToString = resultGroupList.join(',')
```
Comment (skshetry) [inline tests/acceptance/stepDefinitions/sharingContext.js]: The variable name is vague. You can just use the same name instead (this is a dynamic langauge :slightly_smiling_face: )
```suggestion
    excludedGroups = excludedGroups.join(',')
```

File: tests/acceptance/pageObjects/sharedWithMePage.js [hunk 6]
```diff
@@ -68,6 +68,20 @@ module.exports = {
...
+     */
+    assertSharedByDesiredUser: function (element, sharer) {
```
Comment (skshetry) [inline tests/acceptance/pageObjects/sharedWithMePage.js]: `Desired` is not necessary here.
```suggestion
    assertSharedByUser: function (element, sharer) {
```

File: tests/acceptance/pageObjects/sharedWithMePage.js [hunk 7]
```diff
@@ -68,6 +68,20 @@ module.exports = {
         .waitForElementVisible(actionLocatorButton)
         .click(actionLocatorButton)
         .waitForOutstandingAjaxCalls()
+    },
+    /**
+     * @param {string} element
+     * @param {string} sharer
+     *
+     * Asserts that the element(file/folder/resource) on the shared-with-me page is shared by the desired user
```
Comment (dpakach) [inline tests/acceptance/pageObjects/sharedWithMePage.js]: Keep these types of docs above the `@param` tags.

**Pull request merged**